robot/EDA/resources/EDA.py

```diff
@@ -381,3 +424,187 @@ def get_namespace_prefix(self, name):
+            locator_menu,
+            error="Contact tab unavailable"
+        )
+        # javascript is being used here because the usual selenium click is highly unstable for this element on MetaCI
+        self.selenium.driver.execute_script("arguments[0].click()", element_menu)
+        time.sleep(1)
+
+        # Sometimes, single click fails. Hence an additional condition to click on it again
+        if not self.check_if_element_exists(locator_tab):
+            self.selenium.driver.execute_script("arguments[0].click()", element_menu)
+            time.sleep(1)
+
+
+    def select_value_from_listbox(self, title, value):
+        """ Selects value from a listbox/dropdown identified by title """
+        locator_title = eda_lex_locators["listbox"]["title"].format(title)
+        locator_value = eda_lex_locators["listbox"]["value"].format(value)
+        self.selenium.click_element(locator_title)
+        time.sleep(1)
+        self.selenium.click_element(locator_value)
+
+    def select_xpath(self, loc, value):
+        """ Selects the correct xpath by checking if it exists on the page
+            from the given list of locator possibilities
+        """
+        locators = eda_lex_locators[loc].values()
+
+        for i in locators:
+            locator = self.format_all(i, value)
+            if self.check_if_element_exists(locator):
+                return locator
+
+        assert "Button with the provided locator not found"
+
+    def wait_for_locator_with_timeout(self, timeout, path, *args):
+        """Waits for specified element to be available
+            by checking every 1 seconds for given number of seconds
+        """
+        locator = self.get_eda_locator(path, *args)
+
+        i = int(timeout)/7
+        while i > 1:
+            try:
+                logger.console("\n" + "Check:" + str(round(i)) + " - Waiting for locator")
+                self.selenium.get_webelement(locator)
+                return True
+            except Exception:
+                time.sleep(1)
+                i -= 1
+
+        self.builtin.log("Timed out waiting for locator with path " + locator)
+        return False
+
+    def wait_for_new_window(self, title):
+        """ Waits for specified window to be available
+            by checking every 1 seconds for 25 times
+        """
+        window_found = False
+
+        for i in range(25):
+            i += 1
+            time.sleep(1)
+            titles = self.selenium.get_window_titles()
+            for j in titles:
+                if j == title:
+                    window_found = True
+                    return window_found
+
+            if window_found:
+                return
+            else:
+                continue
+
+        self.builtin.log("Timed out waiting for window with title " + title)
+        return window_found
+
+    def shift_to_default_content(self):
+        self.selenium.driver.switch_to.default_content()
```

Comment:
This is missing documentation.  Also, the name isn't very descriptive when used in a test. It looks like you call this shortly after calling `Open EDA Settings Tab menu item` in a test. Would it make sense to rename this to be `Close EDA Settings Tab` or something like that?

Reply:
I included documentation.  It is needed and intentional and worded correctly.